attach/nv_attach_impl/pass/ptxpass_core/src/core.cpp

```diff
@@ -0,0 +1,435 @@
+#include "ptxpass/core.hpp"
+#include <cstdio>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+
+#include "json.hpp"
+
+using nlohmann::json;
+
+namespace ptxpass
+{
+
+static std::vector<std::regex>
+compileRegexList(const std::vector<std::string> &patterns)
+{
+	std::vector<std::regex> out;
+	out.reserve(patterns.size());
+	for (const auto &p : patterns) {
+		out.emplace_back(p, std::regex::ECMAScript);
+	}
+	return out;
+}
+
+static bool jsonGetBool(const json &j, const char *key, bool defVal)
+{
+	if (!j.is_object())
+		return defVal;
+	if (!j.contains(key))
+		return defVal;
+	try {
+		return j.at(key).get<bool>();
+	} catch (...) {
+		return defVal;
+	}
+}
+
+static std::string jsonGetString(const json &j, const char *key,
+				 const std::string &defVal)
+{
+	if (!j.is_object())
+		return defVal;
+	if (!j.contains(key))
+		return defVal;
+	try {
+		return j.at(key).get<std::string>();
+	} catch (...) {
+		return defVal;
+	}
+}
+
+static int jsonGetInt(const json &j, const char *key, int defVal)
+{
+	if (!j.is_object())
+		return defVal;
+	if (!j.contains(key))
+		return defVal;
+	try {
+		return j.at(key).get<int>();
+	} catch (...) {
+		return defVal;
+	}
+}
+
+PassConfig JsonConfigLoader::loadFromFile(const std::string &path)
```

Comment:
Can we use https://github.com/nlohmann/json?tab=readme-ov-file#arbitrary-types-conversions for deserialization?